drivers/pinctrl/pinctrl_wch_00x_afio.c

```diff
@@ -46,14 +46,11 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintp
 		regs->CFGLR = (regs->CFGLR & ~(0x0F << (pin * 4))) | (cfg << (pin * 4));
 
 		if (pins->output_high) {
-			regs->OUTDR |= BIT(pin);
-			regs->BSHR |= BIT(pin);
+			regs->BSHR = BIT(pin);
```

Comment:
+1 to dropping the `BSHR |=`

Looking at the datasheet figure 7.1, the output direction control is after the alternate function logic, so I'm not sure about dropping the OUTDR.

Reply:
Sorry, by dropping I mean changing from |= to =

Reply:
Ah, my mistake.